core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -176,6 +190,25 @@ private class ClientEndpoint(
       } else if (!Utils.responseFromBackup(message)) {
         System.exit(-1)
       }
+
+    case DriverStatusResponse(found, state, _, _, _) =>
+      if (found) {
+        state.get match {
+          case DriverState.FINISHED | DriverState.FAILED |
+               DriverState.ERROR | DriverState.KILLED =>
+            logInfo(s"State of $submittedDriverID is ${state.get}, " +
+              s"exiting spark-submit JVM.")
+            System.exit(0)
+          case _ =>
+            Thread.sleep(REPORT_DRIVER_STATUS_INTERVAL)
+            logInfo(s"State of $submittedDriverID is ${state.get}, " +
```

Comment:
nit: s"State of driver $submittedDriverID ..."

Reply:
Since status polling will happen every second, I'm afraid logs can be too verbose. We can log it after a constant polling times, e.g. log every 60 times.

Reply:
This would produce too much logs, please consider using 'logDebug'

Reply:
Thanks, I have changed it to use `logDebug`.